service/history/transfer_queue_active_task_executor.go

```diff
@@ -400,16 +400,18 @@ func (t *transferQueueActiveTaskExecutor) processCloseExecution(
 		}
 	}
 
-	err = t.processParentClosePolicy(
-		ctx,
-		namespaceName.String(),
-		&workflowExecution,
-		children,
-	)
-
-	if err != nil {
-		// This is some retryable error, not NotFound or NamespaceNotFound.
-		return err
+	// process parentClosePolicy except when the execution was reset. In case of reset, we need to keep the children around so that we can reconnect to them.
+	// TODO (Chetan): update this condition as new reset policies are added. For now we keep all children since "Reconnect" is the only policy available.
+	if !executionInfo.WorkflowWasReset {
```

Comment:
I think this has the same issue as we talked about [here](https://github.com/temporalio/temporal/pull/6694/files#r1828480606). We should use `reset_run_id` instead.

WorkflowWasReset will be set for the current run, not the base run. 

And looks like the bug for nexus callback has not been fixed yet? (cc @bergundy ). Once fixed we can deprecate the WorkflowWasReset field.

Reply:
Good point. Fixed it.